code/mathematical_algorithms/src/coprime_numbers/coprime_numbers.c

```diff
@@ -1,32 +1,37 @@
-// Part of Cosmos by OpenGenus Foundation
-
-#include<stdio.h>
-
-int 
-hcf(int a, int h) 
-{
-    int temp; 
-    while (1) {
-        temp = a % h;
-        if (temp == 0)
-        	return h;
-        a = h;
-        h = temp;
-    }
-} 
-
-int 
-main() 
-{
-    int c, d, gcd; 
-    printf("Enter two Natural Numbers\n");
-    scanf("%d %d", &c, &d);
-    gcd = hcf(c, d);
+#include <stdio.h>
```

Comment:
Comments "// Part of Cosmos by OpenGenus Foundation" should be a part of updated code.

There should not be a new line at the end of the file.

Explain the detailed improvement as a part of the pull request description.